### DISCUSSION TITLE
Matrix rain sometimes starts in rainbow mode

### DISCUSSION THREAD

**Discussion by francofusco** [rocksdanister/lively#1749] [General]
**Describe the bug**
When booting the PC, sometimes the "Matrix Rain" effect is launched with rainbow colors instead of the default green.

**To Reproduce**
Unfortunately I have not been able to determine the exact reason. When I boot the PC, the effect is usually working fine - the rain is green. Sometimes however, the effect is using the "rainbow mode".

**Expected behavior**
The effect should show green rain.

**Screenshots/Video**
N.A.

**Desktop (please complete the following information):**
 - OS: Windows 11 Home
- Which wallpaper was running at the time if any: Matrix Rain Customizable

**Additional context**
I think it is worth mentioning that even when the bug happens, if I open the _Customize_ dialog the _Rainbow_ checkbox is **unchecked**. Not sure if this helps in any way, but if I check it, nothing happens, and if I uncheck it, the matrix becomes violet and I need to change the font color for the effect to become green. My solution for now was to press the _Restore default_ button.

**Log file (Important)**
[lively_log_20230607_085314.zip](https://github.com/rocksdanister/lively/files/11673596/lively_log_20230607_085314.zip)

**Comment by rocksdanister** [comment]
In `script.js` line 8 try setting `rainbow: false,`

**Reply by francofusco** [reply]
Thanks for the reply :)

Since my reporting, I think I got this issue at every reboot. It looks like lively "forgets" about the stored settings?

Anyway, I just updated `script.js` as you suggested, changing also `wavecolor` to be the default `(0, 255, 70)` green shade, otherwise, I am fairly confident it would be purple rain time!

![image](https://github.com/rocksdanister/lively/assets/15146986/c5f9d73b-9bf6-45ab-9913-79e67bfe10ed)

Do you think the effect will be reset (meaning that I will have to manually update it again) when installing an update? I haven't really gone inside the code so I have no idea how it works :P

**Comment by valfrid-ly** [comment]
@francofusco , I tried this solution and now it doesn't start with rainbow color but start with a different color (usually purple).

In my specific case, I have 2 monitors and the external one starts correctly and the laptop screen starts with a different color.

**Reply by francofusco** [reply]
Do you mean that you changed `rainbow: true` into `rainbow: false` inside `script.js`? I did that and in addition I changed `wavecolor` to `(0, 255, 70)` (a dark green shade) and it works well for me. I have a dual monitor setup, one with DP and the other with HDMI. Hope this is enough for you as well! :)

**Reply by valfrid-ly** [reply]
I changed rainbow to false but I didn't change the color manually in the script. As the interface let us to setup a color, all monitors should start with the selected color.

**Reply by francofusco** [reply]
Yes, I agree that it should be the case. But I believe that the issue is precisely that, sometimes, the script ignores what’s been configured in the interface and uses the default parameters in the script itself.

**Reply by valfrid-ly** [reply]
That's why it needs to be fixed

**Reply by rocksdanister** [reply]
Not sure why this is happening, logs don't show any error and other wallpapers don't have this issue.

This wallpaper is quite simple so confused 🤔

**Reply by pvogt** [reply]
If it's any help, I use Lively in my Windows VMware Workstation; and occasionally, when I resize my VMware Workstation window either by clicking full screen or dragging the edges (doing this changes my Windows screen resolution), the Matrix Rain will go rainbow color instead of the colors I'm using.

So, maybe changing / setting the screen resolution triggers or resets something in the coding.
